routes/articles.js

```diff
@@ -0,0 +1,129 @@
+const express = require('express')
+let router = express.Router()
+const Article = require('../models/article')
+const {isLoggedIn, checkArticleOwnership} = require('../middleware')
+
+function escapeRegex(text) {
+    return text.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, "\\$&");
+}
+
+//INDEX ROUTE -- Show all articles
+router.get('/', (req, res) => {
+    if(req.query.search && req.xhr) {
+        const regex = new RegExp(escapeRegex(req.query.search), 'gi')
+        Article.find({name: regex}, (err, articles) => {
```

Comment:
Shouldn't we set some limit of findings per page, so we won't get DDoSed by person requiring whole db?

Reply:
See issue #13